contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -392,10 +406,13 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
       NodeOperator memory nodeOperator = s_nodeOperators[node.nodeOperatorId];
       if (!isOwner && msg.sender != nodeOperator.admin) revert AccessForbidden();
 
-      bool nodeExists = s_nodes[node.p2pId].signer != address(0);
+      bool nodeExists = bytes32(s_nodes[node.p2pId].signer) != bytes32("");
       if (!nodeExists) revert InvalidNodeP2PId(node.p2pId);
 
-      if (node.signer == address(0)) revert InvalidNodeSigner();
+      if (
+        bytes32(node.signer) == bytes32("") ||
+        (s_nodes[node.p2pId].signer != node.signer && s_encodedNodeSignerAddresses.contains(node.signer))
```

Comment:
Need to handle the case of updating a node when their signer address is not updated